src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,22 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.task.TaskList;
+
+import java.io.IOException;
+
+public abstract class Command {
+
+    protected boolean isByeCommand;
+
+    public Command() {
+        this.isByeCommand = false;
+    }
+
+    public abstract String execute(Storage storage, TaskList taskList) throws IOException;
```

Comment:
I like the usage of abstract methods to fit this command context where there are multiple commands, each having their own `execute` implementation